File: java/src/main/java/it/giannibombelli/wsc2026/payment/domain/payment/ProviderReference.java
package it.giannibombelli.wsc2026.payment.domain.payment;

import it.giannibombelli.wsc2026.common.utils.Require;

import java.util.UUID;

public record ProviderReference(UUID value) {
    public ProviderReference {
        Require.requireArgument(value, "providerReference");
    }
}

File: typescript/src/payment/domain/payment/providerReference.ts
// Riferimento opaco: ha significato solo per il provider, payment non lo interpreta.

import { requireArgument } from '@/common/utils/requireArgument.js';
import { Uuid } from '@/common/domain/primitive/uuid.js';

export class ProviderReference {
  readonly value: Uuid;

  constructor(value: Uuid) {
    requireArgument(value, 'providerReference');
    this.value = value;
  }

  toString(): string {
    return this.value.value;
  }
}
